pkg/azure/aks/aks.go

```diff
@@ -88,10 +132,115 @@ func (c *Collector) CollectMetrics(_ chan<- prometheus.Metric) float64 {
 	return 0
 }
 
+func (c *Collector) populateMachinePricesCache() error {
+	c.cacheLock.Lock()
+	defer c.cacheLock.Unlock()
+	c.logger.Info("populating machine prices cache")
+
+	c.MachineStore.machineMapLock.RLock()
+	defer c.MachineStore.machineMapLock.RUnlock()
+
+	for vmName, vmInfo := range c.MachineStore.MachineMap {
+		price, err := c.PriceStore.getPriceInfoFromVmInfo(vmInfo)
+		if err != nil {
+			return err
+		}
+
+		c.MachinePricesCache[vmName] = price
+	}
+
+	return nil
+}
+
+// TODO - BREAK INTO CPU AND RAM
+func (c *Collector) getMachinePrices(vmName string) (float64, error) {
+	// Cache Hit
+	c.cacheLock.RLock()
+	if price, ok := c.MachinePricesCache[vmName]; ok {
+		c.cacheLock.RUnlock()
+		return price, nil
+	}
+	c.cacheLock.RUnlock()
+
+	// Cache Miss
+	c.cacheLock.Lock()
+	defer c.cacheLock.Unlock()
+
+	vmInfo := c.MachineStore.getVmInfoByVmName(vmName)
+	price, err := c.PriceStore.getPriceInfoFromVmInfo(vmInfo)
+	if err != nil {
+		return 0.0, err
+	}
+
+	return price, nil
+}
+
 // Collect satisfies the provider.Collector interface.
 func (c *Collector) Collect(ch chan<- prometheus.Metric) error {
-	// TODO - implement
-	c.logger.LogAttrs(c.context, slog.LevelInfo, "TODO - implement AKS collector Collect method")
+	c.logger.Info("collecting metrics")
+	now := time.Now()
+
+	eg := new(errgroup.Group)
+	if now.After(c.machineStoreNextPopulationTime) {
+		eg.Go(func() error {
```

Comment:
Suggestion: Log out that the machine store needs to be populated, and then log out the duration it took to populate.

Reply:
Ignore the suggestion, you're doing that in the called method 🤦🏻